core/templates/pages/contributor-dashboard-page/services/translate-text.service.ts

```diff
@@ -146,18 +152,15 @@ export class TranslateTextService {
       contentType: contentType,
       interactionId: interactionId,
       ruleType: ruleType
-    };
+    }as TranslatableItem;
```

Comment:
This typecasting shouldn't be done. Types in the interface `TranslatableItem` are different from ones present in the constructor. Properties `text`, `interactionId` & `ruleType` should have common types all throughout this file along with proper comments on why `null` is added as a type.